packages/exoflex/src/components/Button/ButtonOpacity.tsx

```diff
@@ -38,19 +39,18 @@ export default function ButtonOpacity(props: ButtonProps) {
   return (
     <TouchableOpacity
       {...otherProps}
+      ref={ref as RefObject<TouchableOpacity>}
```

Comment:
I'm not sure if passing `ref` directly will work or not.
This will need a bit of testing and see if this needs use forwardRef instead.

For now I think this is OK, just make sure to not release this before testing whether the ref works or not.

Reply:
Got it, will test it.

Reply:
@ikusa @darcien

About this problem, I think I'll open up a discussion in here. So the initial reason why I add this changes was because the parent exoflex `Button` component has props with type `ButtonProps` which inherit `PaperButton` types. 

`PaperButton` props type also inherit `PaperSurface` type, which has the `ref` prop with type `React.RefObject<View>`

Our exoflex Button has 2 type: `ButtonOpacity` & `ButtonRipple`, which will rendered based on `useRipple` props. 

With `ButtonRipple`, which will render `TouchableRipple` from `react-native-paper`, it seems the passed `ref` with type `<React.RefObject<View>>` from `ButtonProps` didn't raise any type error. Turns out `TouchableRipple` inherits it's prop type from RN `Pressable` and also has the `forwaredRef React.RefAttributes<View>`.

But with `ButtonOpacity`, which will render `TouchableOpacity` from `react-native`, it raise ref type error. Because the desired type was `LegacyRef<TouchableOpacity>` but supplied with `RefObject<View>`. That's why I passing the ref like that and typecast it as `RefObject<TouchableOpacity>` to fulfilled the required type.

As for using the `forwardRef`, the same problem arise when I need to type the `ref` from using `forwardRef` on Parent Button component. 

What is the best solution to handle this?

Reply:
It's been a while since I worked on this area, so it might be a bit hard to give exact code solution.

From a glance, we should be able to resolve this type conflict by migrating away from TouchableOpacity and use Pressable instead. That way the ref is always a View ref.

The cons: we need to mimic the old TouchableOpacity using Pressable to keep the same behavior. Unless we want to change the default button behavior, and just do a breaking changes.

Reply:
@ikusa @darcien 

Got it! Changes on using `Pressable` are in this commit: https://github.com/kodefox/infra/pull/713/commits/e46eaf323b3da7ff861c4b2412060b2e644ee284

I added animated opacity to make the feedback looks like `TouchableOpacity` and it seems to work as intended.


https://user-images.githubusercontent.com/25154563/216543775-798ac977-fc63-4187-a6d9-9798e40f086e.mp4

Reply:
Looks great 👍 

You might want to put the animated value in a ref and memoize the fade transition functions.

Also, probably it's not a good idea to override the onPressIn/onPressOut from other props with our fade animation. Will need to extend it so we also call the onPressIn/onPressOut from the props.

Reply:
Got it! Applied in here: https://github.com/kodefox/infra/pull/713/commits/4971cb9dd6c72cb62e64d49525a6fb8931ca65bb